### PR TITLE
feat: add auth0 to hubspot project

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

File: auth0_to_hubspot/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: Auth0 to HubSpot
```
Comment (daabr) [inline auth0_to_hubspot/README.md]: What do you think about "Copy Auth0 Users to HubSpot"? Still short, but clearer.

File: auth0_to_hubspot/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+description: Add new Auth0 users to HubSpot as contacts
```
Comment (daabr) [inline auth0_to_hubspot/README.md]:
```suggestion
description: Periodically add new Auth0 users to HubSpot as contacts
```

File: auth0_to_hubspot/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+integrations: ["auth0", "hubspot"]
+categories: ["Office Automation"]
+---
+
+# Auth0 to HubSpot
+
+This project adds new Auth0 users to HubSpot as contacts.
```
Comment (daabr) [inline auth0_to_hubspot/README.md]:
```suggestion
# Auth0 to HubSpot

This project adds new Auth0 users to HubSpot as contacts, on a recurring basis.
```

File: auth0_to_hubspot/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+1. Fetch new Auth0 users from the last `HOURS` hours.
+2. Create HubSpot contacts for each user.
+
+## Deployment & Configuration
+
+### Cloud Usage (Recommended)
+
+1. Initialize your connections through the UI
+2. (Optional) Configure the `HOURS` variable to set the time range for new users to fetch.
+> [!NOTE]
```
Comment (daabr) [inline auth0_to_hubspot/README.md:24]:
```suggestion

> [!NOTE]
```

File: auth0_to_hubspot/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+---
+
+# Auth0 to HubSpot
+
+This project adds new Auth0 users to HubSpot as contacts.
+
+## How It Works
+
+1. Fetch new Auth0 users from the last `HOURS` hours.
+2. Create HubSpot contacts for each user.
```
Comment (daabr) [inline auth0_to_hubspot/README.md]:
```suggestion
1. Fetch new Auth0 users from the last `HOURS` hours
2. Create HubSpot contacts for each user
```

File: auth0_to_hubspot/program.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+auth0 = auth0_client("auth0_conn")
+hubspot = hubspot_client("hubspot_conn")
+
+LOOKUP_HOURS = int(os.getenv("HOURS"))
+
+
+def check_for_new_users(event):
+    """Workflow entrypoint.
+    Looks up new Auth0 users in the last `HOURS` hours and adds them to HubSpot as contacts.
```
Comment (daabr) [inline auth0_to_hubspot/program.py]:
```suggestion
    """Workflow entrypoint.

    Looks up new Auth0 users in the last `HOURS` hours and adds them to HubSpot as contacts.
```

File: auth0_to_hubspot/program.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import os
+
+from autokitteh.auth0 import auth0_client
+from autokitteh.hubspot import hubspot_client
+from hubspot.crm.contacts import SimplePublicObjectInput
+
+auth0 = auth0_client("auth0_conn")
+hubspot = hubspot_client("hubspot_conn")
+
+LOOKUP_HOURS = int(os.getenv("HOURS"))
```
Comment (daabr) [inline auth0_to_hubspot/program.py]: Constants should usually be above global variables, not below

File: auth0_to_hubspot/program.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    response = auth0.users.list(q=query, search_engine="v3")
+    add_new_users(response["users"])
+
+
+def _get_time_range(hours):
+    """Calculate start and end times for user lookup."""
+    now = datetime.datetime.utcnow()
+    start_time = now - datetime.timedelta(hours=hours)
+
+    return {"start": start_time.isoformat() + "Z", "end": now.isoformat() + "Z"}
```
Comment (daabr) [inline auth0_to_hubspot/program.py]: It's nice that "start" and "end" are explicit, but I think the query template in line 21 is clear enough ("X to Y") and using a dictionary only makes it look more complicated than it is.

A better and more common approach: return a tuple:

```
start, end = _get_time_range(LOOKUP_HOURS)
```

**Review by daabr — APPROVED** [review]

File: auth0_to_hubspot/README.md (hunk 1 above)
Comment (pashafateev) [reply auth0_to_hubspot/README.md]: I couldn't think of a better name, so I think your idea is great.

File: auth0_to_hubspot/program.py (hunk 7 above)
Comment (pashafateev) [reply auth0_to_hubspot/program.py]: Done

File: auth0_to_hubspot/program.py (hunk 8 above)
Comment (pashafateev) [reply auth0_to_hubspot/program.py]: Nice! Good suggestion. Done.

**Pull request merged**